src/Classes/Utils/SpatialGrid.ts

```diff
@@ -0,0 +1,91 @@
+import { ICoordinates } from "../../Interfaces/ICoordinates";
+import { IDimension } from "../../Interfaces/IDimension";
+import { Particle } from "../Particle";
+import { Utils } from "./Utils";
+
+/* This class essentially works by interpreting all particles on the screen as a grid.
+Grid cells are determined by dividing the width and height by the cell size. so 1920 / 10 = 19 cells of width
+Particles are pushed into their respective cells with the same method.
+
+For small particles this is great, but i expect there will need to be allowances for larger particles.
+See https://i.stack.imgur.com/1Gx63.png.
+
+ Hence the addition of the radius option for selecting objects in adjacent cells.
+This also allows for broader detection for things like line drawing
+*/
+
+export class SpatialGrid {
+    private cellSize: number;
+    private widthSegment: number;
+    private heightSegment: number;
+    private grid: object[][][] = [];
+
+    // Cut the grid up into a 2d array with a 3rd dimesion holding the data.
+    constructor(canvas: IDimension, cellSize = 20) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [expected parameter: 'cellSize' to have a typedef](https://app.codacy.com/manual/ar3s/tsparticles/pullRequest?prid=5343253)